### ISSUE TITLE
Pkg.gc not working on windows due to string interpolation

### ISSUE THREAD

**Issue by feanor12** [JuliaLang/Pkg.jl#1681]
I reported a bug to julia, but I think should be listed here. 

https://github.com/JuliaLang/julia/issues/34781

**Comment by StefanKarpinski** [comment]
Possible problem is that TOML uses Julia's own `repr` function to escape strings but TOML format doesn't support all the escapes that Julia does. Is there some kind of weird or invalid unicode going on in there? From the example it doesn't look like it. Can you share the TOML file that it's choking on?

**Comment by feanor12** [comment]
Here is part of that file.
I should not contain anything funny, but parsing still fails.

```
julia> Pkg.TOML.parsefile("error.txt")
ERROR: Pkg.TOML.ParserError(24, 25, "unknown string escape: `\\x24`")
Stacktrace:
 [1] parse(::IOStream) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.3\Pkg\ext\TOML\src\TOML.jl:37
 [2] #open#271(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(open), ::typeof(Pkg.TOML.parse), ::String, ::Vararg{String,N} where N) at .\io.jl:298
 [3] open at .\io.jl:296 [inlined]
 [4] parsefile(::String) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.3\Pkg\ext\TOML\src\TOML.jl:50
 [5] top-level scope at REPL[3]:1
```

[error.txt](https://github.com/JuliaLang/Pkg.jl/files/4218954/error.txt)

**Comment by StefanKarpinski** [comment]
It's that `$` in there: Julia wants to escape it but TOML doesn't recognize `\$` as a valid escape.

**Comment by feanor12** [comment]
I updated to julia 1.4 and the error is still present.
I tested with the error.txt and with calling gc on my system.
How can I check the version of Pkg used by the installation?

**Comment by fredrikekre** [comment]
> How can I check the version of Pkg used by the installation?

Its the same as the Julia version.

**Comment by StefanKarpinski** [comment]
To clarify, the fix prevents writing the bad TOML file in the first place; if you already have bad TOML like you do, you'll have to fix it manually, but going forward newer Julia versions won't produce bad TOML. So to fix that file, delete the backslash before all instances of `$` in strings.

**Comment by feanor12** [comment]
I tried it with 1.4 an a clean file. It still happens.

1. delete manifest_usage.toml and the other files in logs just to be sure
2. run julia 

Here ist the toml file that was produced: [manifest_usage.toml.txt](https://github.com/JuliaLang/Pkg.jl/files/4370779/manifest_usage.toml.txt)

```
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.4.0 (2020-03-21)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia>

(@v1.4) pkg> gc
     Active manifests:
     Active artifacts:
    Deleted no artifacts or packages

(@v1.4) pkg>

(@v1.4) pkg> activate "U:\jupyter workdir\misc\student-t"
 Activating environment at `U:\jupyter workdir\misc\student-t\Project.toml`

(student-t) pkg> activate
 Activating environment at `C:\Users\mpus\.julia\environments\v1.4\Project.toml`

(@v1.4) pkg> gc
ERROR: Pkg.TOML.ParserError(24, 25, "unknown string escape: `\\x24`")

...and 1 more exception(s).

Stacktrace:
 [1] parse(::Base.GenericIOBuffer{Array{UInt8,1}}) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\ext\TOML\src\TOML.jl:37
 [2] parse at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\ext\TOML\src\TOML.jl:44 [inlined]
 [3] (::Pkg.API.var"#collect_usage!#83")(::Dict{String,Dates.DateTime}, ::String) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\src\API.jl:353
 [4] gc(::Pkg.Types.Context; collect_delay::Dates.Day, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\src\API.jl:372
 [5] gc at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\src\API.jl:330 [inlined] (repeats 2 times)
 [6] do_cmd!(::Pkg.REPLMode.Command, ::REPL.LineEditREPL) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\src\REPLMode\REPLMode.jl:400
 [7] do_cmd(::REPL.LineEditREPL, ::String; do_rethrow::Bool) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\src\REPLMode\REPLMode.jl:381
 [8] do_cmd at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\src\REPLMode\REPLMode.jl:376 [inlined]
 [9] (::Pkg.REPLMode.var"#24#27"{REPL.LineEditREPL,REPL.LineEdit.Prompt})(::REPL.LineEdit.MIState, ::Base.GenericIOBuffer{Array{UInt8,1}}, ::Bool) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\src\REPLMode\REPLMode.jl:542
 [10] #invokelatest#1 at .\essentials.jl:712 [inlined]
 [11] invokelatest at .\essentials.jl:711 [inlined]
 [12] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\REPL\src\LineEdit.jl:2354
 [13] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\REPL\src\REPL.jl:1055
 [14] run_repl(::REPL.AbstractREPL, ::Any) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\REPL\src\REPL.jl:206
 [15] (::Base.var"#764#766"{Bool,Bool,Bool,Bool})(::Module) at .\client.jl:383
 [16] #invokelatest#1 at .\essentials.jl:712 [inlined]
 [17] invokelatest at .\essentials.jl:711 [inlined]
 [18] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at .\client.jl:367
 [19] exec_options(::Base.JLOptions) at .\client.jl:305
 [20] _start() at .\client.jl:484

julia> using Pkg

julia> cd(raw"C:\Users\mpus\.julia\logs")

julia> Pkg.TOML.parsefile("manifest_usage.toml")
ERROR: Pkg.TOML.ParserError(24, 25, "unknown string escape: `\\x24`")

...and 1 more exception(s).

Stacktrace:
 [1] parse(::IOStream) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\ext\TOML\src\TOML.jl:37
 [2] open(::typeof(Pkg.TOML.parse), ::String, ::Vararg{String,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at .\io.jl:298
 [3] open at .\io.jl:296 [inlined]
 [4] parsefile(::String) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Pkg\ext\TOML\src\TOML.jl:50
 [5] top-level scope at REPL[8]:1

julia>
```

**Comment by StefanKarpinski** [comment]
Ok, I'll investigate further.

**Comment by StefanKarpinski** [comment]
Mystery solved: this was fixed after we branched 1.4 and never backported. This is fixed on master, just not in the 1.4 release.
